BookingProject/BookingWebAPI/BookingWebApi.Infrastructure/Data/ApartamentRepository.cs

```diff
@@ -55,5 +66,139 @@ public async Task<PageResultResponse<Apartament>> GetApartamets(ApartamentFilter
+            }
+        }
+
+        public async Task<Result<decimal>> GetMedianArea()
+        {
+            await using var connection = _context.Database.GetDbConnection();
+
+            try
+            {
+                var sql = GetSqlQuery("MedianArea", SqlFilePath.StatisticApartmentScripts);
+
+                var result = await connection.QuerySingleAsync<decimal>(sql);
+                return Result<decimal>.Success(result);
+            }
+            catch (Exception ex)
+            {
+                return Result<decimal>.Failure(ex.Message);
+            } 
+        }
+
+        public async Task<Result<List<TotalAreaCountBySourceDto>>> GetTotalAreaCountBySourceCompany()
+        {
+            await using var connection = _context.Database.GetDbConnection();
+
+            try
+            {
+                var sql = GetSqlQuery("TotalAreaAndCountBySourceCompany", SqlFilePath.StatisticApartmentScripts);
+
+                var result = await connection.QueryAsync<TotalAreaCountBySourceDto>(sql);
+                return Result<List<TotalAreaCountBySourceDto>>.Success(result.ToList());
+            }
+            catch (Exception ex)
+            {
+                return Result<List<TotalAreaCountBySourceDto>>.Failure(ex.Message);
+            }
+        }
+
+        public async Task<Result<bool>> UpsertCustomData(int apartamentId, string customData)
+        {
+            await using var connection = _context.Database.GetDbConnection();
+
+            try
+            {
+                var sql = GetSqlQuery("Upsert", SqlFilePath.UpsertApartamentCustomDataScript);
+
+                var result = await connection.ExecuteAsync(sql,new {ApartamentId = apartamentId, CustomData = customData});
+                if(result == 0)
+                {
+                    return Result<bool>.Failure("Problems with sql code");
+                }
+                return Result<bool>.Success(true);
+            }
+            catch (Exception ex)
+            {
+                return Result<bool>.Failure(ex.Message);
+            }
+        }
+
+        public async Task<bool> ApartamentExist(int apartamentId)
+        {
+            var apartament = await _context.Apartaments.FindAsync(apartamentId);
+
+            if (apartament == null) return false;
+            return true;
+        }
+        private string GetSqlQuery(string resourceName,string sqlPath)
+        {
+            if(_sqlCashe.TryGetValue(resourceName, out var cashedSql))
+            {
+                return cashedSql;
+            }
+
+            var resourceManager = new ResourceManager(sqlPath, typeof(ApartamentRepository).Assembly);
+            var sql = resourceManager.GetString(resourceName);
+            if(string.IsNullOrEmpty(sql))
+            {
+                throw new Exception($"SQL query {resourceName} not found");
+            }
+
+            _sqlCashe[resourceName] = sql;
+            return sql;
+        }
```

Comment:
It should be accessible via script name as static fields in the app
ChatGPT said:
ChatGPT
To make SQL scripts accessible as static fields in your application, you can use the Resources designer class generated by the .resx file. Here's how:

Steps:
Add SQL Scripts to .resx:

Open your .resx file (e.g., Resources.resx) in Visual Studio.
Add SQL scripts as string resources with meaningful names (e.g., GetUsersQuery, InsertUserQuery).
Access the Scripts via Static Fields:

The Resources designer class automatically generates static properties for each resource.
Access these properties directly in your code.
Example:
1. Add SQL Scripts to Resources.resx:
Key: GetUsersQuery
sql
Copy code
SELECT * FROM Users WHERE IsActive = 1;
Key: InsertUserQuery
sql
Copy code
INSERT INTO Users (Name, Email) VALUES (@Name, @Email);
2. Access the Scripts in Code:
csharp
Copy code
using YourNamespace.Properties;

class Program
{
    static void Main()
    {
        // Access SQL scripts as static properties
        string getUsersQuery = Resources.GetUsersQuery;
        string insertUserQuery = Resources.InsertUserQuery;

        // Use the scripts
        Console.WriteLine("Get Users Query: " + getUsersQuery);
        Console.WriteLine("Insert User Query: " + insertUserQuery);
    }
}
Key Notes:
Namespace for Resources:

By default, it is YourNamespace.Properties.Resources.
Ensure you include the correct namespace.
Generated Static Properties:

The .resx file generates a Resources class with static properties for all added entries.
No Need for ResourceManager:

The Resources class internally uses ResourceManager, so you don’t need to manage it manually.
Pros:
Static Access: Easy and type-safe access to SQL scripts.
Readability: Scripts are organized and accessible via meaningful names.
No Manual Coding: The Resources class is auto-generated.
Cons:
Compilation Dependency: Changes to scripts require recompilation.
Size Limitation: Large scripts increase the size of the assembly.

Reply:
Fixed